File: 01_language_comparison/data_structure_manipulation/stack.java
public class Stack<T> {
  private Node<T> top;

  // Node class for linked list
  private static class Node<T> {
    T data;
    Node<T> next;
    Node(T data) {
      this.data = data;
    }
  }

  // Add element on top
  public void push(T data) {
    Node<T> node = new Node<>(data);
    node.next = top;
    top = node;
  }

  // Remove and return top
  public T pop() {
    if (isEmpty()) throw new RuntimeException("Stack Underflow");
    T data = top.data;
    top = top.next;
    return data;
  }

  // Return top without removing
  public T peek() {
    if (isEmpty()) throw new RuntimeException("Stack is empty");
    return top.data;
  }

  public boolean isEmpty() {
    return top == null;
  }
}

// Example:
// Stack<Integer> s = new Stack<>();
// s.push(1); s.pop(); s.peek();

File: 01_language_comparison/data_structure_manipulation/stack.js
class Stack {
    constructor() {
      this.items = [];
    }
  
    // Add element to top
    push(element) {
      this.items.push(element);
    }
  
    // Remove and return top element
    pop() {
      if (this.isEmpty()) throw new Error("Stack Underflow");
      return this.items.pop();
    }
  
    // Return top element without removing
    peek() {
      if (this.isEmpty()) throw new Error("Stack is empty");
      return this.items[this.items.length - 1];
    }
  
    isEmpty() {
      return this.items.length === 0;
    }
}
  
// Example usage:
// const stack = new Stack();
// stack.push(10); stack.pop(); stack.peek();